src/comparator.js

```diff
@@ -47,59 +48,75 @@ function normalise (entry: any) {
-    Bucket: bucket,
-    Prefix: sfprefix
+    Bucket: fulfilment.downloadFolder.bucket,
+    Prefix: fulfilment.downloadFolder.prefix,
+    Delimeter: '/'
   }).promise()
-  const sfkeys = sfresp.Contents.map(r => { return r.Key.slice(sfprefix.length) }).filter(notEmpty)
+  const sfkeys = sfresp.Contents.map(r => { return r.Key.slice(fulfilment.downloadFolder.prefix.length) }).filter(notEmpty)
 
-  console.log(guprefix)
+  console.log('Fetching existing fulfilment file list from S3', fulfilment.uploadFolder.bucket, fulfilment.uploadFolder.prefix)
   const guresp = await s3.listObjectsV2({
-    Bucket: bucket,
-    Prefix: guprefix
+    Bucket: fulfilment.uploadFolder.bucket,
+    Prefix: fulfilment.uploadFolder.prefix,
+    Delimeter: '/'
   }).promise()
-  const gukeys = guresp.Contents.map(r => { return r.Key.slice(guprefix.length) }).filter(notEmpty)
-  console.log(logprefix)
+  const gukeys = guresp.Contents.map(r => { return r.Key.slice(fulfilment.uploadFolder.prefix.length) }).filter(notEmpty)
+
+  let logFolder: S3Folder = {
+    bucket: fulfilment.downloadFolder.bucket,
+    prefix: `${fulfilment.downloadFolder.prefix}logs/`
+  }
+
+  console.log('Fetching existing comparison file list from S3', logFolder.bucket, logFolder.prefix)
+
   const logresp = await s3.listObjectsV2({
-    Bucket: bucket,
-    Prefix: logprefix
+    Bucket: logFolder.bucket,
+    Prefix: logFolder.prefix,
+    Delimeter: '/'
   }).promise()
-  const logkeys = logresp.Contents.map(r => { return r.Key.slice(guprefix.length) }).filter(notEmpty)
 
-  let sfDates: Array<moment> = sfkeys.map(salesforceDate).filter(notEmpty)
-  let guDates: Array<moment> = gukeys.map(outputDate).filter(notEmpty)
+  const logkeys = logresp.Contents.map(r => { return r.Key.slice(logFolder.prefix.length) }).filter(notEmpty)
+
+  let sfFiles: Array<Filename> = sfkeys.map(extractFilename).filter(notEmpty)
+  let guFiles: Array<Filename> = gukeys.map(extractFilename).filter(notEmpty)
+  let logFiles: Array<Filename> = logkeys.map(extractFilename).filter(notEmpty).filter(inFuture)
 
-  let logDates: Array<moment> = logkeys.map(logDate).filter(notEmpty).filter(inFuture)
   // Check all future dated logfiles every time we run, just to make sure we haven't missed an update.
 
-  console.log('Found the following salesforce fulfilments', sfDates.map(d => d.format(OUTPUT_DATE_FORMAT)))
-  console.log('Found the following fulfilments', guDates.map(d => d.format(OUTPUT_DATE_FORMAT)))
-  console.log('Found the following logs', logDates.map(d => d.format(OUTPUT_DATE_FORMAT)))
+  console.log('Found the following salesforce fulfilments', sfFiles.map(f => f.filename))
+  console.log('Found the following fulfilments', guFiles.map(f => f.filename))
+  console.log('Found the following logs', logFiles.map(f => f.filename))
+
+  let sfMap = new Map(sfFiles.map(f => [f.date, f]))
+  let combined = new Map(sfFiles.filter(f => sfMap.has(f.date)).map(f => [f.date, {salesforce: sfMap.get(f.date), fulfilment: f}]))
```

Comment:
what is that filter supposed to do here? Unless I'm misunderstanding something it seems like it would never filter out anything